src/main/java/com/mineboundteam/minebound/magic/OffensiveSpells/FireOffensiveSpell.java

```diff
@@ -0,0 +1,165 @@
+import com.mineboundteam.minebound.item.armor.ArmorTier;
+import com.mineboundteam.minebound.magic.ActiveSpellItem;
+import com.mineboundteam.minebound.magic.MagicType;
+import com.mineboundteam.minebound.magic.SpellType;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.SmallFireball;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.ForgeConfigSpec;
+import net.minecraftforge.fml.event.config.ModConfigEvent;
+import org.jetbrains.annotations.Nullable;
+
+import java.text.DecimalFormat;
+import java.util.List;
+
+public class FireOffensiveSpell extends ActiveSpellItem {
+
+    private final int manaCost;
+    private final int fireDistance;
+    private final double fireDamage;
+    private final boolean igniteBlocks;
+    private final boolean shootFireball;
+    private final int fireballManaCost;
+
+    public FireOffensiveSpell(Properties properties, FireOffensiveSpellConfig config) {
+        super(properties, config.LEVEL, MagicType.FIRE, SpellType.OFFENSIVE);
+
+        this.manaCost = config.MANA_COST.get();
+        this.fireDistance = config.FIRE_DISTANCE.get();
+        this.fireDamage = config.FIRE_DAMAGE.get();
+        this.igniteBlocks = config.IGNITE_BLOCKS.get();
+        this.shootFireball = config.SHOOT_FIREBALL.get();
+        this.fireballManaCost = config.FIREBALL_MANA_COST.get();
+    }
+
+    @Override
+    public void use(ItemStack stack, Level level, Player player) {
+        if (shootFireball) {
+            float yRot = player.getYRot();
+            float xRot = player.getXRot();
+            double x = (0 - Math.sin(yRot * Math.PI / 180f));
+            double y = (0 - Math.sin(xRot * Math.PI / 180f));
+            double z = (Math.cos(yRot * Math.PI / 180f));
+            SmallFireball smallfireball = new SmallFireball(level, player.getX(), player.getEyeY() - 1.15d, player.getZ(), x, y, z);
+            level.addFreshEntity(smallfireball);
+            level.playSound(null, player.getX(), player.getEyeY(), player.getZ(), SoundEvents.FIRECHARGE_USE, SoundSource.PLAYERS, 1f, 1f);
+            reduceMana(fireballManaCost, player);
+        }
+    }
+
+    @Override
+    public void onUsingTick(ItemStack stack, Level level, Player player, int tickCount) {
+        float yRot = player.getYRot();
+        float xRot = player.getXRot();
+        double x = (0 - Math.sin(yRot * Math.PI / 180f));
+        double y = (0 - Math.sin(xRot * Math.PI / 180f));
+        double z = (Math.cos(yRot * Math.PI / 180f));
+        for (int i = 0; i <= 360; i += 30) {
+            level.addFreshEntity(new FireProjectile(level,
+                    player.getX(),
```

Comment:
Do you think it would be relatively easy to shift this left or right depending on the using hand? So the origin looks like it's from where the player is casting it?

Reply:
I don't know how easy it will be, I'd have to take a shot at it to see which I can do.

Reply:
you could probably hardcode the values; Do you want me to merge Light Utility so the animation works in singleplayer? You'll have to refactor the classes with your UsingTick changes

Reply:
Yeah, you can merge Light Utility.

Reply:
Just tried `ItemStack.isSameItemSameTags(getSelectedSpell(player, PlayerSelectedSpellsProvider.PRIMARY_SPELL), stack)` which has the opposite problem, it's always `true` even if the secondary spell is the one currently being used.

Reply:
Actually, I have an idea, I'll post an update once I've tried it.

Reply:
Alright, it worked. Required updating the parameters for all the using functions in `ActiveSpellItem`, but it works like a charm.